### DISCUSSION TITLE
表头+表格这种组合，没办法识别出表头标题

### DISCUSSION THREAD

**Discussion by gaogaomie** [RapidAI/TableStructureRec#105] [Q&A]
我测试的例子用的是人行征信报告，其实某种意义上，它算是表格，但在使用ocr识别和html组装的过程中，它会把表头标题给忽略掉了，这种情况求大佬解答下，该怎么解决

![demo](https://github.com/user-attachments/assets/d9a024e9-3caf-4e50-a200-8e1424e15f49)
![纸质版2](https://github.com/user-attachments/assets/59d45cbf-5a51-45bd-a05b-069eff55574a)
![params](https://github.com/user-attachments/assets/10c07119-39ff-47eb-b606-e61f07ceb26a)

**Comment by Joker1212** [comment]
可以先进行版面识别，会将表格区域和文字区域分开，再分别进行表格识别和文字ocr，也可以先版面识别后进行整页ocr识别，可能更快，整页ocr的结果给到表格识别模型，会进行box匹配的

**Reply by Joker1212** [reply]
或者格式固定的话，直接后面再用ocr添加一下也可以

**Reply by Joker1212** [reply]
可以试试rapidLayout,我试了这个模型    layout_engine = RapidLayout(model_type="doclayout_docstructbench", conf_thres=0.2)
，应该是能区分表格标题和内容框的，分别按box截图，再进行ocr识别和表格识别就行了，可能预先做个识别框去重，保留分数高的
<img width="598" alt="image" src="https://github.com/user-attachments/assets/d5d27ece-4bf6-4e37-bf9c-2a84ca27f7f4" />

**Reply by gaogaomie** [reply]
好的，我试试，谢谢大佬解答

**Reply by gaogaomie** [reply]
【先版面识别后进行整页ocr识别，整页ocr的结果给到表格识别模型，会进行box匹配的】
这个思路我觉得挺不错的，大佬你这个有参考的代码或者项目吗？

**Reply by Joker1212** [reply]
整页ocr识别就是把这个图片给RapidOCR识别，然后直接把这个结果丢给表格模块就行了，readme中有RapidOCR配合传入ocr结果的代码，表格模型只会匹配识别框重叠的，要提高效率你可以自己先根据版面识别的表格区域bbox过滤一下ocr的bbox
